vslib/src/sai_vs_init.cpp

```diff
@@ -439,6 +399,232 @@ sai_status_t create_hostif_traps()
     return SAI_STATUS_SUCCESS;
 }
 
+sai_status_t create_scheduler_group_tree(
+        _In_ const std::vector<sai_object_id_t>& sgs,
+        _In_ sai_object_id_t port_id)
+{
+    SWSS_LOG_ENTER();
+
+    sai_attribute_t attrq;
+
+    std::vector<sai_object_id_t> queues;
+
+    // in this implementation we have 20 queues per port
+    // (10 in and 10 out), which will be assigned to schedulers
+    uint32_t queues_count = 20;
+
+    queues.resize(queues_count);
+
+    attrq.id = SAI_PORT_ATTR_QOS_QUEUE_LIST;
+    attrq.value.objlist.count = queues_count;
+    attrq.value.objlist.list = queues.data();
+
+    CHECK_STATUS(vs_generic_get(SAI_OBJECT_TYPE_PORT, port_id, 1, &attrq));
+
+    // schedulers groups: 0 1 2 3 4 5 6 7 8 9 a b c
+
+    // tree index
+    // 0 = 1 2
+    // 1 = 3 4 5 6 7 8 9 a
+    // 2 = b c (bug on brcm)
+
+    // 3..c - have both QUEUES, each one 2
+
+    // sg 0 (2 groups)
+    {
+        sai_object_id_t sg_0 = sgs.at(0);
```

Comment:
Depends how to look at this, since this is one off, if written correctly it will never fail, and if some actions will lead to throw it will need to be corrected right away since this logic is executed in vs tests when doing make check, then is auto tested